src/components/common/MemberProfileButton.tsx

```diff
@@ -44,31 +46,93 @@ const BorderedItem = styled(List.Item)`
     margin-right: -12px;
   }
 `
+const StyledCollapseIconWrapper = styled(Box)`
+  && {
+    margin: auto 0;
+  }
+`
 
-export const CustomNavLinks: React.VFC = () => {
-  const { navs } = useNav()
+export const CollapseNavLinks: React.VFC<{ nav: any }> = ({ nav }) => {
+  const { isOpen, onToggle } = useDisclosure()
+
+  const ListItem = (
+    <List.Item key={nav.id} style={{ cursor: 'pointer' }}>
+      {nav.icon ? <Icon type={nav.icon} className="mr-2" /> : <BlankIcon className="mr-2" />}
+      {nav.label}
+    </List.Item>
+  )
 
   return (
     <>
-      {navs
-        .filter(nav => nav.block === 'header')
-        .map((nav, idx) => {
-          const ListItem = (
-            <List.Item key={idx} style={{ cursor: 'pointer' }}>
-              {nav.icon ? <Icon type={nav.icon} className="mr-2" /> : <BlankIcon className="mr-2" />}
-              {nav.label}
-            </List.Item>
-          )
-
-          return nav.external ? (
-            <a key={idx} href={nav.href} target="_blank" rel="noopener noreferrer">
+      {nav.href ? (
+        <>
+          {nav.external ? (
+            <a key={nav.id} href={nav.href} target="_blank" rel="noopener noreferrer">
               {ListItem}
             </a>
           ) : (
-            <Link key={idx} to={nav.href}>
+            <Link key={nav.id} to={nav.href}>
               {ListItem}
             </Link>
-          )
+          )}
+        </>
+      ) : (
+        <>
+          <Box d="flex" justifyContent="space-between" cursor="pointer" onClick={onToggle}>
+            <Box>{ListItem}</Box>
+            <StyledCollapseIconWrapper>
+              {isOpen ? <ChevronDownIcon w="16px" /> : <ChevronRightIcon w="16px" />}
+            </StyledCollapseIconWrapper>
+          </Box>
+          <Collapse in={isOpen} animateOpacity style={{ background: '#f7f8f8', margin: '0 -12px' }}>
+            {nav.subNavs.map(
+              (v: {
+                id: string
+                block: string
+                position: number
+                label: string
+                icon: string | null
+                href: string
+                external: boolean
+                locale: string
+                tag: string | null
+              }) => {
```

Comment:
replace `v` with `subNav` as parameter
not need to infer types here if use `AppNavProps` above